walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,120 @@
 
 package com.google.sps;
 
+import java.util.ArrayList;
 import java.util.Collection;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    private static final int START_OF_DAY = TimeRange.START_OF_DAY;
+    private static final int END_OF_DAY = TimeRange.END_OF_DAY;
+    private static final TimeRange WHOLE_DAY = TimeRange.WHOLE_DAY;
+    private static final Comparator<TimeRange> ORDER_BY_START = TimeRange.ORDER_BY_START;
+
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+
+        // Array of queries represenitng times to meet.
+        List<TimeRange> query = new ArrayList<>();
+        
+        int numberOfEvents = events.size();
+        
+        int timeOfStart = START_OF_DAY;
+        
+        int desiredDuration = (int) request.getDuration();
+        
+        //  Event counter used whilst iterating.
+        int eventCounter = 0;
+        
+        int eventStart = 0;
+        int eventEnd = 0;
+
+        // Set of stirngs representing unique employees set to attend this meeting.
+        Set<String> attendees = new Set();
+
+        // Collection of string reprenting who is to attend the requedted meeting
+        Collection<String> requestAttendees = request.getAttendees();
+        
+        // Boolean used to indictae whter or not the attendee should be ignored.
+        boolean shouldIgnore = true;
+
+        // Check if duration of event is longer than a whole day and if so return empty.
+        if ((int)request.getDuration() > WHOLE_DAY.duration()) {
+            return query;
+        }
+
+        // Check if there are any events happening at all if so return whole day query.
+        if (numberOfEvents == 0) {
+            // Adds a 24 hour time range to query.
+            query.add(TimeRange.WHOLE_DAY);
+            return query;
+        }
+        // Iterating throguh input event collection. 
+        for (Event event : events.sort(event.ORDER_BY_START)) {
+
+            // Collection of the pople who are to attend meeting.
+            attendees = event.getAttendees();
+
+            //Ignore event only if attendees of this event are not needed in requested meeting
+            for (String attendee : attendees) {
+                // If this attendee is required
+                if (requestAttendees.contains(attendee)) {
+                    // You shouldn't ignore this attendee. 
+                    shouldIgnore = false;
+                    break;
+                } 
+                // If at the last event.
+                if (eventCounter == numberOfEvents-1) {
```

Comment:
nit: spaces around the - sign. 

Consider setting up the google format pre-commit hook if you haven't already. It will catch a lot of these for you.

Reply:
I just added https://github.com/google/google-java-format to my pom.xml I think that should work right ?